sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,265 @@
+    :ref:`User Guide <stacking_transformer>`.
+
+    Parameters
+    ----------
+    estimator : predictor
+        The estimator to be blended.
+
+    cv : int, cross-validation generator or an iterable, optional (default=3)
+        Determines the cross-validation splitting strategy to be used for
+        generating features to train the next layer on the stacked ensemble or,
+        more specifically, during ``fit_transform``.
+
+        Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross-validation,
+        - integer, to specify the number of folds.
+        - An object to be used as a cross-validation generator.
+        - An iterable yielding train/test splits.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass,
+        :class:`sklearn.model_selection.StratifiedKFold` is used. In all
+        other cases, :class:`sklearn.model_selection.KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    method : string, optional (default='auto')
+        This method will be called on the estimator to produce the output of
+        transform. If the method is ``auto``, will try to invoke, for each
+        estimator, ``predict_proba``, ``decision_function`` or ``predict``
+        in that order.
+
+    n_jobs : int, optional (default=1)
+        Number of jobs to be passed to ``cross_val_predict`` during
+        ``fit_transform``.
+    """
+    def __init__(self, estimator, cv=3, method='auto', n_jobs=1):
+        self.estimator = estimator
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, **fit_params):
+        """Fit the estimator.
+
+        This should only be used in special situations. Read more in the
+        :ref:`User Guide <stacking_transformer>`.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        self : object
+
+        """
+        self.estimator_ = clone(self.estimator)
+        self.estimator_.fit(X, y, **fit_params)
+        return self
+
+    def _method_name(self):
+        if self.method == 'auto':
+            if getattr(self.estimator_, 'predict_proba', None):
+                method = 'predict_proba'
+            elif getattr(self.estimator_, 'decision_function', None):
+                method = 'decision_function'
+            else:
+                method = 'predict'
+        else:
+            method = self.method
+
+        return method
+
+    def transform(self, *args, **kwargs):
```

Comment:
Is there other way to avoid `*args` and `*kwargs`.
@jnothman are the different method having some keywords?

It would be at least good to have `X` to be forced.